### ISSUE TITLE
Remove INSFVMomentumFriction

### ISSUE THREAD

**Issue by lindsayad** [idaholab/moose#20857]
## Reason
We originally added the object to fake a representation of a heat exchanger. Now that we have `PINSFVMomentumFriction`, we can use that instead

## Design
Delete `INSFVMomentumFriction`

## Impact
Should be little impact on users because they can replace with `PINSFVMomentumFriction`. Positive impact for us as maintainers because we can just focus on `PINSFVMomentumFriction`

**Comment by kyriv1980** [comment]
Is this issue resolved?? @lindsayad @grmnptr

**Comment by lindsayad** [comment]
it's not resolved really since the kernel is still in the module. We did modify the form of `PINSFVMomentumFriction` in https://github.com/idaholab/moose/pull/21230 so that it now appears to be very similar to `INSFVMomentumFriction` ... so in theory one might think it would be easy to remove now. However, I believe I encountered at least one case where the removal is non-trivial ... I think there was at least one input somewhere that was solving inviscid flow but still had friction terms applied. In such a case it would be non-trivial to do the replacement since we multiply the Darcy coefficient by the viscosity in the `PINSFVMomentumFriction` kernel
